$NetBSD: patch-configure.in,v 1.3 2024/07/16 11:40:16 adam Exp $

- Sun/NetBSD audio support
- Install extra headers needed by portmixer in Audacity for OSS.
- Don't use -lpthread
- IRIX: pa_sgi is no longer in the source; build without a native host API.

--- configure.in.orig	2021-04-01 00:35:42.000000000 +0000
+++ configure.in
@@ -41,6 +41,10 @@
             AS_HELP_STRING([--with-oss], [Enable support for OSS @<:@autodetect@:>@]),
             [with_oss=$withval])
 
+AC_ARG_WITH(sun,
+            AS_HELP_STRING([--with-sun], [Enable support for Sun/NetBSD Audio @<:@autodetect@:>@]),
+            [with_sun=$withval])
+
 AC_ARG_WITH(asihpi,
             AS_HELP_STRING([--with-asihpi], [Enable support for ASIHPI @<:@autodetect@:>@]),
             [with_asihpi=$withval])
@@ -141,6 +145,10 @@
         AC_CHECK_LIB(ossaudio, _oss_ioctl, have_libossaudio=yes, have_libossaudio=no)
     fi
 fi
+have_sun=no
+if test "x$with_sun" != "xno"; then
+    AC_CHECK_HEADERS([sys/audioio.h], [have_sun=yes], [have_sun=no])
+fi
 have_jack=no
 if test "x$with_jack" != "xno"; then
     PKG_CHECK_MODULES(JACK, jack, have_jack=yes, have_jack=no)
@@ -278,6 +286,7 @@
         SHARED_FLAGS="$LIBS -dynamiclib $mac_arches $mac_sysroot $mac_version_min"
         CFLAGS="-std=c99 $CFLAGS $mac_arches $mac_sysroot $mac_version_min"
         OTHER_OBJS="src/os/unix/pa_unix_hostapis.o src/os/unix/pa_unix_util.o src/hostapi/coreaudio/pa_mac_core.o src/hostapi/coreaudio/pa_mac_core_utilities.o src/hostapi/coreaudio/pa_mac_core_blocking.o src/common/pa_ringbuffer.o"
+        INCLUDES="$INCLUDES pa_mac_core.h"
         PADLL="libportaudio.dylib"
         ;;
 
@@ -362,13 +371,9 @@
         dnl The 'dmedia' library is needed to read the Unadjusted System Time (UST).
         dnl
         AC_CHECK_LIB(pthread, pthread_create, , AC_MSG_ERROR([IRIX posix thread library not found!]))
-        AC_CHECK_LIB(audio,   alOpenPort,     , AC_MSG_ERROR([IRIX audio library not found!]))
-        AC_CHECK_LIB(dmedia,  dmGetUST,       , AC_MSG_ERROR([IRIX digital media library not found!]))
 
-        dnl See the '#ifdef PA_USE_SGI' in file pa_unix/pa_unix_hostapis.c
-        dnl which selects the appropriate PaXXX_Initialize() function.
-        dnl
-        AC_DEFINE(PA_USE_SGI,1)
+        dnl pa_sgi (the IRIX AL host API) is gone from the source: build
+        dnl without a native host API.
 
         CFLAGS="$CFLAGS -I\$(top_srcdir)/src/os/unix"
 
@@ -376,11 +381,11 @@
         dnl
         THREAD_CFLAGS="-D_REENTRANT"
 
-        OTHER_OBJS="pa_sgi/pa_sgi.o src/os/unix/pa_unix_hostapis.o src/os/unix/pa_unix_util.o"
+        OTHER_OBJS="src/os/unix/pa_unix_hostapis.o src/os/unix/pa_unix_util.o"
 
         dnl SGI books say -lpthread should be the last of the libs mentioned.
         dnl
-        LIBS="-lm -ldmedia -laudio -lpthread"
+        LIBS="-lm -lpthread"
         PADLL="libportaudio.so"
         SHARED_FLAGS=""
         ;;
@@ -411,6 +416,7 @@
 
         if [[ "$with_oss" != "no" ]] ; then
            OTHER_OBJS="$OTHER_OBJS src/hostapi/oss/pa_unix_oss.o"
+           INCLUDES="$INCLUDES pa_unix_oss.h"
            if [[ "$have_libossaudio" = "yes" ]] ; then
                    DLL_LIBS="$DLL_LIBS -lossaudio"
                    LIBS="$LIBS -lossaudio"
@@ -418,6 +424,11 @@
            AC_DEFINE(PA_USE_OSS,1)
         fi
 
+        if [[ "$have_sun" = "yes" ] && [ "$with_sun" != "no" ]] ; then
+           OTHER_OBJS="$OTHER_OBJS src/hostapi/sun/pa_unix_sun.o"
+           AC_DEFINE(PA_USE_SUN,1)
+        fi
+
         if [[ "$have_asihpi" = "yes" ] && [ "$with_asihpi" != "no" ]] ; then
            LIBS="$LIBS -lhpi"
            DLL_LIBS="$DLL_LIBS -lhpi"
@@ -425,8 +436,8 @@
            AC_DEFINE(PA_USE_ASIHPI,1)
         fi
 
-        DLL_LIBS="$DLL_LIBS -lm -lpthread"
-        LIBS="$LIBS -lm -lpthread"
+        DLL_LIBS="$DLL_LIBS -lm ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}"
+        LIBS="$LIBS -lm ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}"
         PADLL="libportaudio.so"
 
         ## support sun cc compiler flags
@@ -505,6 +516,7 @@
      *)
 	AC_MSG_RESULT([
   OSS ......................... $have_oss
+  Sun ......................... $have_sun
   JACK ........................ $have_jack
 ])
         ;;
